packages/eslint-plugin/src/rules/ban-ts-comment.ts

```diff
@@ -192,6 +201,14 @@ export default createRule<[Options], MessageIds>({
           }
           const { directive, description } = match;
 
+          if (
+            directive === 'nocheck' &&
+            firstStatement &&
+            isPositionEarlierThan(firstStatement.loc.start, comment.loc.start)
```

Comment:
```suggestion
            firstStatement.loc.start.line <= comment.loc.start.line
```

[Bug] If `// @ts-nocheck` comment is placed on the line before actual code, but its `loc.start.column` is greater than `firstStatement.loc.start.column`, the `nocheck` directive is not reported, even though this is a valid TS pragma and it effectively suppresses all TS errors.

```ts
 /// @ts-nocheck
const a: true = false
```

[playground link](https://deploy-preview-10046--typescript-eslint.netlify.app/play#ts=5.5.2&fileType=.tsx&code=AQejwAQFwZwWgHYHsDGALApig1gKBUgjFMAIYBcwUATgK4bAC8wAZqQDYwZA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AIwEMnb9layAewC2wxC3RRE0aIOiRwYAL4glQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

I don't think we need to compare `loc` columns here. Since singleline comments should be placed either on their own line or at the end of a line.
And if `// @ts-nocheck` is placed at the end of a line, it's no longer a valid TS pragma.

Reply:
Oh oops. 🙈 

I understand that checking columns alone is a bug, my intention was to not report the following:

```ts
/* @ts-nocheck - should not be reported */ const a = 1;
```

The function I was going to write was something like:

```ts
function isPositionEarlierThan(
  a: TSESTree.Position,
  b: TSESTree.Position,
): boolean {
  if (a.line > b.line) {
    return false;
  }
  
  return a.line === b.line && a.column < b.column;
}
```

But I can see the block comment with `ts-nocheck` doesn't disable type-checking [and the lint rule doesn't report this already](https://github.com/typescript-eslint/typescript-eslint/blob/896c7312dfe8374312df9404cb11db8378cd091c/packages/eslint-plugin/src/rules/ban-ts-comment.ts#L109-L110).

I've changed my code accordingly, thanks 🙏

Reply:
I think it's worth including my example in the test suite as well. That way we can prevent regressions in the future.